storage/database/sharded_database.go

```diff
@@ -70,11 +70,25 @@ func newShardedDB(dbc *DBConfig, et DBEntryType, numShards uint) (*shardedDB, er
 	sdbBatchTaskCh := make(chan sdbBatchTask, numShards*2)
 	sdbLevelDBCacheSize := dbc.LevelDBCacheSize / int(numShards)
 	sdbOpenFilesLimit := dbc.OpenFilesLimit / int(numShards)
+	sdbRocksDBCacheSize := GetDefaultRocksDBConfig().CacheSize / uint64(numShards)
+	sdbRocksDBMaxOpenFiles := GetDefaultRocksDBConfig().MaxOpenFiles / int(numShards)
+	if dbc.RocksDBConfig != nil {
+		sdbRocksDBCacheSize = dbc.RocksDBConfig.CacheSize / uint64(numShards)
+		if dbc.RocksDBConfig.MaxOpenFiles == -1 {
+			sdbRocksDBMaxOpenFiles = dbc.RocksDBConfig.MaxOpenFiles
```

Comment:
I am not sure we need to allow `-1` configuration. If it causes memory leak, I think we'd better restrict the usage of it

Reply:
Agreed. I have updated not to allow unlimited max open files.